price_service/server-rust/build.rs

```diff
@@ -0,0 +1,73 @@
+use std::{
+    env,
+    path::PathBuf,
+    process::Command,
+};
+
+fn main() {
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+    let out_var = env::var("OUT_DIR").unwrap();
+
+    // Clone the Wormhole repository, which we need to access the protobuf definitions for Wormhole
+    // P2P message types.
+    let _ = Command::new("git")
+        .args([
+            "clone",
+            "https://github.com/wormhole-foundation/wormhole",
+            "wormhole",
+        ])
+        .output()
+        .expect("failed to execute process");
+
+    // Move the tools directory to the root of the repo because that's where the build script
+    // expects it to be, paths get hardcoded into the binaries.
+    let _ = Command::new("mv")
+        .args(["wormhole/tools", "tools"])
+        .output()
+        .expect("failed to execute process");
+
+    // Move the protobuf definitions to the src/network directory, we don't have to do this
+    // but it is more intuitive when debugging.
+    let _ = Command::new("mv")
+        .args([
+            "wormhole/proto/gossip/v1/gossip.proto",
+            "src/network/p2p.proto",
+        ])
+        .output()
+        .expect("failed to execute process");
+
+    // Build the protobuf compiler.
+    let _ = Command::new("./build.sh")
+        .current_dir("tools")
+        .output()
+        .expect("failed to execute process");
+
+    // Make the protobuf compiler executable.
+    let _ = Command::new("chmod")
+        .args(["+x", "tools/bin/*"])
+        .output()
+        .expect("failed to execute process");
+
+    // Generate the protobuf definitions. See buf.gen.yaml to see how we rename the module for our
+    // particular use case.
+    let _ = Command::new("./tools/bin/buf")
+        .args(["generate", "--path", "src"])
+        .output()
+        .expect("failed to execute process");
```

Comment:
Alternatively couldn't we just copy the proto definition and keep it in our repo and build the go file ourself without relying on wh tools?

Reply:
Yes It's an option, I preferred this for a couple of reasons:

- It matches the `Makefile` in Wormhole
- The Wormhole repo doesn't have the compiled proto files, so to include them we'd need to do a few manual steps that aren't reflected in the repo. There's not a lot but they're not intuitive IMO:
  - Fetch upstream Wormhole.
  - Build the tools in our repo to capture the right hardcoded paths (ugly Go compiler thing).
  - Create a patch `buf.gen.yaml` file (which differs from Wormhole's) and generate p2p.pb.go. 
- I think we're on the way to wanting Proto definitions of our own types, and we'd have a similar process.

Really the wormhole clone is the ugliest part of this I agree, I think instead of pinning a version as you said above we should just make building the tools part of our own repo and copy the `.proto` file. I'll do it in a second PR.

Reply:
yeah i think the plan ^ makes sense.